### ISSUE TITLE
[New App Shortcut]: Jio Cinema and Sony Liv

### ISSUE THREAD

**Issue by clintferns2911** [PRProd/HA-Firemote#359]
### App Name

Jio Cinema and Sony Liv

### Device Platform

Amazon Fire

### Application Details

<html>
<body>

restored: true source_list: com.amazon.tv.launcher device_class: tv supported_features: 22961
--

</body>
</html>

### Optional Comments

Not sure if I did this correctly? 
these are the URLS for the services, they have a web version and I also have them installed on my fire tv
https://www.jiocinema.com/
https://www.sonyliv.com/

**Comment by PRProd** [comment]
@clintferns2911 ~

This is a good start, but unfortunately you didn't provide the required information in the **Application Details** section.  Additionally, since you put two requests into one issue (usually I prefer one per issue, but this is OK) I will need you to provide that information twice - once for each application.

[Click here to review the instructions](https://github.com/PRProd/HA-Firemote/wiki/Providing-Details-for-App-Launcher-Requests#for-fire-tv-chromecast-nvidia-shield-xiaomi-mi-and-other-android-based-devices)

I tried to download the apps you requested, but I'm not sure that I have the correct versions. which is why I need you to do this procedure on your end.  

For example, after launching **Jio Cinema** and completing the steps through step 9, this highlighted section is what I need to know from you:

![image](https://github.com/PRProd/HA-Firemote/assets/25486599/889a5079-4442-4c3d-90a4-d0e81c360aa5)

This is how you would reply:
```
adb_response: mSurface=Surface(name=)/@0xabfb729
      mSurface=Surface(name=com.jio.media.stb.ondemand/com.v18.voot.ui.JVHomeActivity)/@0x8d9b037
  mCurrentFocus=Window{23f76cf u0 com.jio.media.stb.ondemand/com.v18.voot.ui.JVHomeActivity}
  mFocusedApp=AppWindowToken{83f9a72 token=Token{8b83b7d ActivityRecord{f17ddd4 u0 com.jio.media.stb.ondemand/com.v18.voot.ui.JVHomeActivity t3218}}}
```

### Conclusion:
What I need next is the **adb_response** for both of your app requests:
 - JioCinema
 - Sony LIV

Thanks

**Comment by PRProd** [comment]
@clintferns2911 - Have you had a chance to look at this yet?

**Comment by clintferns2911** [comment]
HI Apologies, I'll try your instructions and will get back to you

**Comment by clintferns2911** [comment]
Hi sorry for the delayed response. I've followed the steps however i can only get this info
<html>
<body>

adb_response: mAnimationIsEntrance=true      mSurface=Surface(name=com.amazon.tv.launcher/com.amazon.tv.launcher.ui.HomeActivity_vNext)/@0x6202058   mCurrentFocus=Window{df06755 u0 com.amazon.tv.launcher/com.amazon.tv.launcher.ui.HomeActivity_vNext}   mFocusedApp=AppWindowToken{e41206c token=Token{af8951f ActivityRecord{6d4f0be u0 com.amazon.tv.launcher/.ui.HomeActivity_vNext t1276}}} hdmi_input: null device_class: tv friendly_name: Fire TV 192.168.0.197 supported_features: 22961 source: com.amazon.tv.launcher entity_picture: /api/media_player_proxy/media_player.fire_tv_192_168_0_197?token=9dbbb2d11309c5da6d7d7653ce88ea73af6c5f087b2b19bcfbbafaea747b2bfd&cache=727aec1df3c6bc65
--



</body>
</html>


I can't get it to display the jio cinema or sony liv details. Although I can open the app on my tv with my remote.

**Comment by clintferns2911** [comment]
Oh wait, I launched it on the tv and then looked back at home assisant and saw this

<html>
<body>

adb_response: mAnimationIsEntrance=true      mSurface=Surface(name=com.amazon.tv.launcher/com.amazon.tv.launcher.ui.HomeActivity_vNext)/@0x6202058   mCurrentFocus=Window{df06755 u0 com.amazon.tv.launcher/com.amazon.tv.launcher.ui.HomeActivity_vNext}   mFocusedApp=AppWindowToken{e41206c token=Token{af8951f ActivityRecord{6d4f0be u0 com.amazon.tv.launcher/.ui.HomeActivity_vNext t1276}}} hdmi_input: null device_class: tv friendly_name: Fire TV 192.168.0.197 supported_features: 22961 source: Jio Cinema entity_picture: /api/media_player_proxy/media_player.fire_tv_192_168_0_197?token=9dbbb2d11309c5da6d7d7653ce88ea73af6c5f087b2b19bcfbbafaea747b2bfd&cache=4044300f08cc78ab source_list: Jio Cinema app_id: com.jio.media.stb.ondemand
--



</body>
</html>

**Comment by clintferns2911** [comment]
for sony liv

adb_response: mAnimationIsEntrance=true      mSurface=Surface(name=com.amazon.tv.launcher/com.amazon.tv.launcher.ui.HomeActivity_vNext)/@0x6202058
  mCurrentFocus=Window{df06755 u0 com.amazon.tv.launcher/com.amazon.tv.launcher.ui.HomeActivity_vNext}
  mFocusedApp=AppWindowToken{e41206c token=Token{af8951f ActivityRecord{6d4f0be u0 com.amazon.tv.launcher/.ui.HomeActivity_vNext t1276}}}
hdmi_input: null
device_class: tv
friendly_name: Fire TV 192.168.0.197
supported_features: 22961
source: com.onemainstream.sonyliv.android
entity_picture: /api/media_player_proxy/media_player.fire_tv_192_168_0_197?token=9dbbb2d11309c5da6d7d7653ce88ea73af6c5f087b2b19bcfbbafaea747b2bfd&cache=646ec7593e90c2a8
source_list: com.onemainstream.sonyliv.android
app_id: com.onemainstream.sonyliv.android

**Comment by clintferns2911** [comment]
I also noticed that the button on the dashboard didn't launch disney plus even though i had it installed on the firestick, i assume cause it's an indian version of dinsey+ so i put that info as well below

adb_response: mAnimationIsEntrance=true      mSurface=Surface(name=com.amazon.tv.launcher/com.amazon.tv.launcher.ui.HomeActivity_vNext)/@0x6202058
  mCurrentFocus=Window{df06755 u0 com.amazon.tv.launcher/com.amazon.tv.launcher.ui.HomeActivity_vNext}
  mFocusedApp=AppWindowToken{e41206c token=Token{af8951f ActivityRecord{6d4f0be u0 com.amazon.tv.launcher/.ui.HomeActivity_vNext t1276}}}
hdmi_input: null
device_class: tv
friendly_name: Fire TV 192.168.0.197
supported_features: 22961
source: Disney+ Hotstar
entity_picture: /api/media_player_proxy/media_player.fire_tv_192_168_0_197?token=9dbbb2d11309c5da6d7d7653ce88ea73af6c5f087b2b19bcfbbafaea747b2bfd&cache=40e251d7229f7549
source_list: Disney+ Hotstar
app_id: in.startv.hotstar

**Comment by clintferns2911** [comment]
I'm really sorry casue I'm very new to all of this and thank you for being so kind and patient with me.

**Comment by PRProd** [comment]
@clintferns2911 ~

Thanks!  These responses are a lot closer to what I was looking for... not quite perfect though.  :smile:  What needs to happen is for you to open the app on your device first, and **after** it's open, *then* you run the command on your Home Assistant system.  No problem, though.  The information you provided does help a little bit.
<br>

## Jio Cinema
I think I have what I need for this one
<br>

## Sony Liv
I think you might be running an old version of this app.  Here is what I found from the information you provided in your answer as compared to the latest version that I found online:
|  | Your Post | I found |
| --- | --- | --- |
| app name | com.onemainstream.sonyliv.android | com.sonyliv |
| download link |  https://apkcombo.com/sonyliv/com.onemainstream.sonyliv.android/ | https://apkpure.com/sony-liv-sports-entertainment/com.sonyliv |
| last updated | Feb 27, 2019 | Mar 22, 2024 |
| Logo | ![image](https://github.com/PRProd/HA-Firemote/assets/25486599/f42a17fd-0e50-4a9c-9ed1-51051a78ee3c)  | ![image](https://github.com/PRProd/HA-Firemote/assets/25486599/273f1e7d-fd1f-453e-b3f7-14fe3b255441) |

Can you check for me to confirm that you have the latest version?  If you do, please launch that app FIRST, then run the command, then post the results for me one more time.
<br>

## Disney+
Based on the information you sent, I don't believe that you are running the [Disney+](https://en.wikipedia.org/wiki/Disney%2B) app.  It looks like you actually have an app called [Disney+ Hotstar](https://en.wikipedia.org/wiki/Disney%2B_Hotstar).  Please open a new [app shortcut request](https://github.com/PRProd/HA-Firemote/issues/new?assignees=PRProd&labels=New+App+Shortcut%2Cenhancement&projects=&template=App_Shortcut_Request.yaml&title=%5BNew+App+Shortcut%5D%3A+Disney%2B%20Hotstar&appname=Disney%2B%20Hotstar) for Disney+ Hotstar

**Comment by PRProd** [comment]
@clintferns2911 ~

I just published Firemote version [v3.2.2](https://github.com/PRProd/HA-Firemote/releases/tag/v3.3.2) which contains an app launch button for Jio Cinema that should work properly, and a button for Sony liv for you to try. (that one  probably won't work)  After updating to the newest Firemote version, if you are unable to see the new updates, please visit this link for help: https://github.com/PRProd/HA-Firemote/wiki/Force-a-Refresh

I'm looking forward to hearing from you soon.

-Doug

**Comment by clintferns2911** [comment]
Nevermind, I had to redownload it. JioCinema works fine! Thanks
SonyLIV doesn't work. What would I need to do?

I also found that fire stick lite 2020 works the best for me, not sure if
you did anything new?

Finally how would I add Disney+ hotstar (Indian version of Disney+)

Also I was planning on getting the latest firestick 4K version from Amazon,
would anything need to be changed with that?

Thanks

Clinton

**Comment by PRProd** [comment]
@clintferns2911 ~

> Nevermind, I had to redownload it. JioCinema works fine! Thanks

Great!  I was actually just writing a response to you about that, so I'm glad you were able to figure it out on your own.
<br>

> SonyLIV doesn't work. What would I need to do?

That was somewhat expected, which is why [I mentioned that it probably wouldn't work](https://github.com/PRProd/HA-Firemote/issues/359#issuecomment-2050627358).   In order to move towards getting it to work, I need you to take a look at [what I wrote about **Sonly liv** in my comments above](https://github.com/PRProd/HA-Firemote/issues/359#issuecomment-2041621560) and provide answers to my questions.
<br>

> I also found that fire stick lite 2020 works the best for me, not sure if
you did anything new?

I didn't do anything new.  _You_ did something new by choosing the correct device model as I had guided you to do [in discussion #360](https://github.com/PRProd/HA-Firemote/discussions/360#discussioncomment-8942335).  When you've set everything properly through the Firemote menus, Firemote does the best it can to be as compatible with your devices as possible.  If you tell Firemote that you are controlling the wrong kind of device, that's where you run into issues.  It's important to get your settings right, so I'm glad you were able to do that.
<br>

> Finally how would I add Disney+ hotstar (Indian version of Disney+)

As I mentioned in my [reply above](https://github.com/PRProd/HA-Firemote/issues/359#issuecomment-2041621560#Disney+):

- Based on the information you sent, I don't believe that you are running the [Disney+](https://en.wikipedia.org/wiki/Disney%2B) app. It looks like you actually have an app called [Disney+ Hotstar](https://en.wikipedia.org/wiki/Disney%2B_Hotstar). Please open a new [app shortcut request](https://github.com/PRProd/HA-Firemote/issues/new?assignees=PRProd&labels=New+App+Shortcut%2Cenhancement&projects=&template=App_Shortcut_Request.yaml&title=%5BNew+App+Shortcut%5D%3A+Disney%2B%20Hotstar&appname=Disney%2B%20Hotstar) for Disney+ Hotstar
<br>

> Also I was planning on getting the latest firestick 4K version from Amazon,
would anything need to be changed with that?

Yes.  When you use a different model, you need to select that model from the list.  Depending on if you get the 4K or the 4K Max, that will determine which one you should use.

![image](https://github.com/PRProd/HA-Firemote/assets/25486599/5b47360d-175d-41c9-a604-787c682bc738)

> [!CAUTION]
As stated in the Firemote Wiki under the section titled [Known Limitations](https://github.com/PRProd/HA-Firemote/wiki#known-limitations) I wrote this **very important** message:
>
>![image](https://github.com/PRProd/HA-Firemote/assets/25486599/6b093cdc-6b47-485f-866b-a01e20c2b7ee)
>
> If you buy one of those two devices listed, you WILL HAVE LAG when using the navigation and some other buttons.  This was your original concern with the device you have now.  If you use one of these two listed devices, the lag will be there and there will be no fix for it.


I hope this helps.  

I look forward to your [new app shortcut request for Disney+ Hotstar](https://github.com/PRProd/HA-Firemote/issues/new?assignees=PRProd&labels=New+App+Shortcut%2Cenhancement&projects=&template=App_Shortcut_Request.yaml&title=%5BNew+App+Shortcut%5D%3A+Disney%2B%20Hotstar&appname=Disney%2B%20Hotstar), and the answers to [my questions about **Sony liv**](https://github.com/PRProd/HA-Firemote/issues/359#issuecomment-2041621560)

-Doug

**Comment by clintferns2911** [comment]
Hello Doug!

For Sony Liv,  this is what it shows, I checked this after I opened the app
on my tv. I hope this is what you need.


source_list: com.onemainstream.sonyliv.android app_id:
com.onemainstream.sonyliv.android source: com.onemainstream.sonyliv.android
adb_response:
***@***.***
mCurrentFocus=Window{eab07f6 u0
com.onemainstream.sonyliv.android/com.sonyliv.ui.home.HomeActivity}
mFocusedApp=AppWindowToken{9ba7622 token=Token{80b77ed
ActivityRecord{598604 u0
com.onemainstream.sonyliv.android/com.sonyliv.ui.home.HomeActivity t1293}}}
hdmi_input: null device_class: tv entity_picture:
/api/media_player_proxy/media_player.fire_tv_192_168_0_197?token=19c63e74fe82ca93b00250f0feeb7dfc817f42b993f43b7ad9ee0806f7b4ccbf&cache=3430e31b5da387af
friendly_name: Fire TV 192.168.0.197 supported_features: 22961


For Disney+Hotstar I submitted a new request

https://github.com/PRProd/HA-Firemote/issues/371

Glad you told me about the issues with the firestick 4K and 4K max, those
we're the devices that I was looking at getting and now will hold on
getting them. The current firestick runs fine, just a bit slow, especially
when it first starts.

I also noticed that when it first starts and if I try to use the firemote
card to control it, it won't allow me to navigate, however if I use my
physical remote and press a left or right button for example and then use
the card, it works fine. Not sure if this is my error or a known thing.


Thanks!

Clinton

**Comment by clintferns2911** [comment]
Oh I also wanted to ask if it was possible to build out my own card?

For example if I only wanted the app launchers, the navigation part and the
volume and home part?

thanks

**Comment by PRProd** [comment]
@clintferns2911,

Hello again Clinton!

> For Sony Liv,  this is what it shows, I checked this after I opened the app
on my tv. I hope this is what you need.

Thank you for answering that part of my question, but I still need more information.  As [stated above](https://github.com/PRProd/HA-Firemote/issues/359#issuecomment-2041621560) I also need you to check to see if you are actually using the newest version of the Sony liv app, and tell me which of the two logos I posted is the correct one in your region for the most updated version of that application.
<br>

> Glad you told me about the issues with the firestick 4K and 4K max, those
we're the devices that I was looking at getting and now will hold on
getting them. The current firestick runs fine, just a bit slow, especially
when it first starts.
>
> I also noticed that when it first starts and if I try to use the firemote
card to control it, it won't allow me to navigate, however if I use my
physical remote and press a left or right button for example and then use
the card, it works fine. Not sure if this is my error or a known thing.

Topic moved to [discussion #360](https://github.com/PRProd/HA-Firemote/discussions/360#discussioncomment-9121830)
<br>

>Oh I also wanted to ask if it was possible to build out my own card?
>
> For example if I only wanted the app launchers, the navigation part and the
volume and home part?

Topic moved to [discussion #360](https://github.com/PRProd/HA-Firemote/discussions/360#discussioncomment-9121830)

**Comment by PRProd** [comment]
Good news @clintferns2911!

I was able to finally find the version of this app that you are using, so I was also able to get the rest of the information I needed to complete your request.  I have released Firemote version [v3.3.3](https://github.com/PRProd/HA-Firemote/releases/tag/v3.3.3) which contains both JioCinema and Sony liv, so I've closed this issue.

After updating to the newest Firemote version, if you are unable to see the new updates on your Firemote card, please visit [this link](https://github.com/PRProd/HA-Firemote/wiki/Force-a-Refresh) for help.

**Comment by PRProd** [comment]
Sorry... I was wrong.  I sent a message saying that the Sony liv issue was complete, but that was an error on my part.

**Comment by PRProd** [comment]
@clintferns2911 ~

We're going to get this working for you, have no fear!  :smile: 

> ...this is my first interaction with a dev and this is all very new and daunting to me. I got your last message and will try to post in the correct place, but if I mess up again, please guide me. I'm still learning :)

Oh wow!  Please don't be intimidated by me or this process in any way, Clinton.  I'm just a random guy who made a thing you have questions about.  :smile:  Asking for assistance is encouraged.  I apologize if I've said or done anything that makes things feel more daunting.

Since your question includes issues with the JioCinema and Sony liv shortcut buttons, this is on topic [[New App Shortcut]: Jio Cinema and Sony Liv #309](https://github.com/PRProd/HA-Firemote/issues/359#top) You've posted this question in the right place:
> So after trying the last update that I got on HA, I can't see the new disney+ hotstar that you mentioned was added. I tried to clear cache by trying ctrl+f5, i also tried a restart of the server and as a final resort even tried to remove firemote and reinstall it through HACS.
>
>I still didn't find the disney+hotstar shortcut, however to my horror i can't see the jio cinema or the sony liv shortcut either (even thought it didn't work)

:rocket: You are on the right track with your troubleshooting steps!  This is absolutely a cache issue!  :rocket: 

Oddly enough though, somehow Home Assistant cache is so incredibly sticky that sometimes it won't clear/refresh through the normal methods, and sometimes it even has this strange ability to revert to something from a long time ago!  While it can be frustrating when this happens, a strong persistent cache is actually a good thing!  It contributes towards a lot of the speed and ease of use we all enjoy while using Home Assistant.

So how do we overcome it?  You've likely seen me post this link before.  It happens to be the most popular article on the Firemote wiki, and it explains multiple ways you can use to try to clear your sticky cache.  You've already done a few of these things, but there are more steps to try, so I encourage you to read it and go through the steps:
https://github.com/PRProd/HA-Firemote/wiki/Force-a-Refresh

Unfortunately, that screen shot you sent me was an image of the HACS store that shows my gitHub repo readme page.  It does not show your dashboard, but that's OK.  This is what it could look like for you when you've solved the cache issue:
![image](https://github.com/PRProd/HA-Firemote/assets/25486599/2e186aad-2dc4-4f11-b4ba-50ded0ada1fd)

---

After you have that resolved, I'm still waiting to hear back from you on this topic:
> ... I still need more information. As [stated above](https://github.com/PRProd/HA-Firemote/issues/359#issuecomment-2041621560) I also need you to check to see if you are actually using the newest version of the Sony liv app, and tell me which of the two logos I posted is the correct one in your region for the most updated version of that application.

After I know that you are indeed using the newest version of the **Sony liv** app, and you tell me which of those two logos is the one you see on your system, I believe that I'll be able to complete this issue.

Thanks!

-Doug

**Comment by PRProd** [comment]
@clintferns2911 ~

Good news, hopefully!  I've just published Firemote version [v3.4.0](https://github.com/PRProd/HA-Firemote/releases/tag/v3.4.0) which I believe should contain working launch buttons for both JioCinema and your version of Sony liv.  I'm going to close this issue as resolved, but if something still doesn't work right for you, please let me know.

Enjoy!

-Doug
